api/src/main/java/org/apache/iceberg/types/AssignIds.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.types;
+
+import java.util.Iterator;
+import java.util.List;
+import java.util.function.Supplier;
+import org.apache.iceberg.Schema;
+import org.apache.iceberg.relocated.com.google.common.collect.Lists;
+
+class AssignIds extends TypeUtil.CustomOrderSchemaVisitor<Type> {
+  private final TypeUtil.GetID getID;
+
+  /**
+   * Replaces the ids in a schema with ids from a base schema, or uses nextId to assign a fresh ids.
```

Comment:
So this is a constructor for a class which can re-assign Id's based on a passed in getter function correct? I think perhaps this logic is now just in the Schema class itself now so maybe the java doc here now has to be more generic to match the impl of this class

Reply:
Yea, removing this javadoc, as its an internal class.  AssignFreshIds, which I based this off, also does not have javadoc.